### ISSUE TITLE
Inconsistency in the BOS token id between tokenizer.encode() and tokenizer.bos_token_id for the WhisperTokenizer

### ISSUE THREAD

**Issue by aladdinpersson** [huggingface/blog#793]
I am using the WhisperTokenizer from the transformers library to encode a string and noticed that the encoded id for the beginning of sequence token (BOS) is different from the bos_token_id provided by the tokenizer. Specifically, when I use tokenizer.encode(""), the encoded id for the BOS token is 50258, but when I access tokenizer.bos_token_id, it is 50257. This seems to be a bug in the code.

Additionally, I noticed that in the finetuning Whisper blog post, the validation loss is increasing over time while the WER is slightly decreasing. This seems like a red flag, but the blog post does not mention this. Can you provide an explanation for this behavior?

**Steps to reproduce:**
```python
from transformers import WhisperTokenizer
tokenizer = WhisperTokenizer.from_pretrained(
    f"openai/whisper-tiny", task="transcribe"
)
encoded_string = tokenizer.encode("")[0]
print(encoded_string) # should print 50258
print(tokenizer.bos_token_id) # should print 50257
```

**Expected outcome:**
The encoded id for the BOS token from `tokenizer.encode("")` and the value of `tokenizer.bos_token_id` should be the same.

**Actual outcome:**
The encoded id for the BOS token from `tokenizer.encode("")` is 50258, but the value of `tokenizer.bos_token_id` is 50257.
